clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsClusterProvider.java

```diff
@@ -36,4 +46,50 @@ public EcsClusterProvider(Cache cacheView) {
   public Collection<EcsCluster> getAllEcsClusters() {
     return ecsClusterCacheClient.getAll();
   }
+
+  public Collection<Cluster> getAllEcsClustersDescription(String account, String region) {
+    List<String> clusters = new ArrayList<>();
+    List<Cluster> response = new ArrayList<>();
+    int clusterCount = 0;
+
+    Collection<EcsCluster> ecsClusters = ecsClusterCacheClient.getAll();
+
+    if (ecsClusters.size() > 0) {
+      for (EcsCluster ecsCluster : ecsClusters) {
+        clusters.add(ecsCluster.getName());
+        clusterCount++;
+
+        if (clusterCount % EcsClusterDescriptionMaxSize == 0) {
+          AmazonECS client = getAmazonEcsClient(account, region);
+          DescribeClustersRequest describeClustersRequest =
+              new DescribeClustersRequest().withClusters(clusters);
+          DescribeClustersResult describeClustersResult =
+              client.describeClusters(describeClustersRequest);
+          for (Cluster cluster : describeClustersResult.getClusters()) {
+            response.add(cluster);
+          }
```

Comment:
Do you think this could be separated out into it's own method that's just responsible for taking a list of cluster names and returning the list of clusters?

Reply:
This particular code is moved into the new method called getDescribeClusters to remove redundancy.